### DISCUSSION TITLE
Network inspector

### DISCUSSION THREAD

**Discussion by mzedeler** [cypress-io/cypress#14998] [Questions and Help]
In the Chrome/Electron network inspector, it seems that all requests, even those that are intercepted by cy.intercept are recorded.

Can someone confirm this?

**Comment by bahmutov** [comment]
Yes, that is true - the `cy.intercept` intercepts the requests after they leave the browser, thus the Network tab still shows them

**Reply by mzedeler** [reply]
Even if they are intercepted with a fixture?

**Comment by bahmutov** [comment]
Yes, even with a fixture

On Tue, Feb 9, 2021 at 10:45 AM Michael Zedeler <notifications@github.com>
wrote:

> Even if they are intercepted with a fixture?
>
> —
> You are receiving this because you commented.
> Reply to this email directly, view it on GitHub
> <https://github.com/cypress-io/cypress/discussions/14998#discussioncomment-353487>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AAQ4BJTNY75YH7I3IZC2BEDS6FJ7ZANCNFSM4XLHEG4A>
> .
>


-- 
Dr. Gleb Bahmutov, PhD

Schedule video chat / phone call / meeting with me via
https://calendly.com/bahmutov
gleb.bahmutov@gmail.com @bahmutov <https://twitter.com/@bahmutov>
https://glebbahmutov.com/ https://glebbahmutov.com/blog
https://github.com/bahmutov

**Reply by mzedeler** [reply]
Ok. Thats surprising - how is that implemented? I am curious because I can see that the Cypress interception code is written in JavaScript, run by the browser (I can even step into it with the debugger), so does that code then somehow communicate something rewriting the network traffic outside the Electron browser?

I am asking because I have an example of a bunch of interceptors that doesn't get called (while others do get called) and I can't figure out what happens.

**Reply by mzedeler** [reply]
I've fired up wireshark and I can see that there seems to be a proxy involved, so I assume that Cypress is setting up this http proxy that does the actual rewriting of requests?

**Reply by bahmutov** [reply]
Yes, Cypress process acts as a proxy for the browser it launches. When the proxy intercepts the network request it sends serialized request back to the browser process via a WebSocket message, triggering the callback. You can step through the callback, and when the result is returned it is sent back to the proxy, where it is written back and gets back to the browser.

**Reply by mzedeler** [reply]
Thanks! This clears up a lot of confusion for me. I was expecting the network interception to be purely client side.

**Reply by mzedeler** [reply]
I am not very good at reading the manual from front to back, whenever I work with a new technology, so if I were to suggest any change to the documentation that could have helped me here, then it would be a reference from the `cy.intercept` manual page to an internals page where the internals of the network interception setup is explained.
